connector/src/main/scala/com/microsoft/kusto/spark/datasource/KustoReader.scala

```diff
@@ -44,29 +44,29 @@ private[kusto] case class KustoReadRequest(sparkSession: SparkSession,
 private[kusto] case class KustoReadOptions(readMode: Option[ReadMode] = None,
                                            shouldCompressOnExport: Boolean = true,
                                            exportSplitLimitMb: Long = 1024)
-
 private[kusto] object KustoReader {
   private val myName = this.getClass.getSimpleName
 
   private[kusto] def singleBuildScan(kustoClient: Client,
                                      request: KustoReadRequest,
                                      filtering: KustoFiltering): RDD[Row] = {
 
-    val filteredQuery = KustoFilter.pruneAndFilter(request.schema, request.query, filtering)
+    val filteredQuery = KustoFilter.pruneAndFilter(KustoSchema(request.schema, Set()), request.query, filtering)
     val kustoResult = kustoClient.execute(request.kustoCoordinates.database,
       filteredQuery,
       request.clientRequestProperties.orNull)
 
     val serializer = KustoResponseDeserializer(kustoResult)
-    request.sparkSession.createDataFrame(serializer.toRows, serializer.getSchema).rdd
+    request.sparkSession.createDataFrame(serializer.toRows, serializer.getSchema.sparkSchema).rdd
   }
 
   private[kusto] def distributedBuildScan(kustoClient: Client,
                                           request: KustoReadRequest,
                                           storage: Seq[KustoStorageParameters],
                                           partitionInfo: KustoPartitionParameters,
                                           options: KustoReadOptions,
-                                          filtering: KustoFiltering): RDD[Row] = {
+                                          filtering: KustoFiltering,
+                                          timespanColumns: Set[String]): RDD[Row] = {
```

Comment:
here as well, kustoScheme? it seems too specific to pass this

Reply:
answered there

Reply:
mostly because it is strange that a function called `distributedBuildScan` will require knowledge of timespanColumns. it will make more sense down the line